### ISSUE TITLE
Add option to remove duplicate instructions

### ISSUE THREAD

**Issue by perliedman** [perliedman/leaflet-routing-machine#100]
For some scenarios, especially bike planning, OSRM spits out the same instruction several times.

![image](https://cloud.githubusercontent.com/assets/1246614/7061906/ed92bdec-de97-11e4-9af4-a47dd24d00ba.png)

Add an option to `L.Routing.OSRM` to remove duplicates, or add some kind of callback option to be able to fix this.

**Comment by TheMarex** [comment]
Best place to fix this would be directly in the OSRM backend.

**Comment by perliedman** [comment]
@TheMarex yes, agreed! This LRM change would only be a workaround for something that will hopefully be fixed in OSRM.

I've looked at the code you mentioned before (https://github.com/Project-OSRM/osrm-backend/blob/master/descriptors/description_factory.hpp#L97), which I guess is relevant to this issue, but still don't feel comfortable to make changes to OSRM. Guess I'll have to dust off my C++ skills.
